tinyagentos/vram_reservation.py

```
@@ -0,0 +1,186 @@
+        Returns ``(free_mb, total_mb)``, or ``(0, 0)`` when no
+        nvidia-smi is available or the probe fails.
+        """
+        try:
+            from tinyagentos.system_stats import read_nvidia_vram
+
+            pair = read_nvidia_vram()
+            if pair is not None:
+                used_mb, total_mb = pair
+                free_mb = max(0, total_mb - used_mb)
+                return free_mb, total_mb
+        except Exception:
```

Ruff (0.15.20)

[warning] Do not catch blind exception: `Exception`

(BLE001)